### PR TITLE
[RC-1453] fx-ify the remote config service component

### PR DESCRIPTION
**Pull request by moth-a**
### What does this PR do?

This PR refactors a bit of remote config service initialization code that was polluting `cmd/agent/common`. There are no functional changes. To accomplish this, the rc service is now being provided as an uber fx component to the agent commands that need it.

### Motivation

First, there is an overarching initiative to migrate to using uber.fx to supply various components throughout the agent.

Second, the remote config service initialization logic was pulled out to `cmd/agent/common` during a prior refactor which is an antipattern, and it did not belong there

Third, as part of the HAMR initiative, we will need to conditionally instantiate/inject a second RC service that communicates with the R2 failover datacenter. Since the RC service depends on an RCTelemetryReporter which can only be initialized/configured once, we really need a singleton telemetry reporter anyways, and uber.fx is the right way to do that regardless. By componentizing everything like this, it sets us up to provide the same telemetry reporter singleton to both services in an upcoming/unrelated PR.

### Describe how to test/QA your changes

- start the cluster agent with RC disabled via `remote_configuration.enabled: false` in the agent config. Ensure the cluster agent starts fine and that remote config is indeed disabled
- start the cluster agent with RC enabled (including a valid API Key etc). Ensure the cluster agent starts fine, and that remote config is indeed enabled
- start the core agent with RC disabled via `remote_configuration.enabled: false` in the agent config. Ensure the cluster agent starts fine and that remote config is indeed disabled
- start the core agent with RC enabled (including a valid API Key etc). Ensure the cluster agent starts fine, and that remote config is indeed enabled

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by ameske — APPROVED** [review]

**Review by L3n41c — APPROVED** [review]
👍🏻 LGTM for files owned by @DataDog/container-integrations.

**Review by ogaca-dd — COMMENTED** [review]
Great PR, added minor comments.

**Comment by timothyalexandersoftware** [comment]
> Great PR, added minor comments.

**Comment by timothyalexandersoftware** [comment]
> Great PR, added minor comments.

@ogaca-dd Thanks for the review! These were super helpful comments for someone who is new to uber fx. I believe I've addressed them all 👍

File: cmd/agent/subcommands/jmx/command.go [hunk 1]
```diff
@@ -138,6 +140,7 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 			fx.Provide(func() demultiplexer.Component { return nil }),
 			fx.Provide(func() inventorychecks.Component { return nil }),
 			fx.Provide(func() packagesigning.Component { return nil }),
+			fx.Provide(func() optional.Option[rcservice.Component] { return optional.NewOption[rcservice.Component](nil) }),
```
Comment (ogaca-dd) [inline cmd/agent/subcommands/jmx/command.go:143]: Note: If the component is never enabled, you can use instead `optional.NewNone[rcservice.Component]()`

File: comp/remote-config/rcservice/rcserviceimpl/rcservice.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+		fx.Provide(newRemoteConfigService),
+	)
+}
+
+// ModuleOptional conditially provides the remote config service.
+func ModuleOptional() fxutil.Module {
+	return fxutil.Component(
+		fx.Provide(newRemoteConfigServiceOptional),
+	)
+}
```
Comment (ogaca-dd) [inline comp/remote-config/rcservice/rcserviceimpl/rcservice.go:39]: As `Module` is never used, you can remove `Module` and rename `ModuleOptional` to `Module` to be consistent with existing code.

File: pkg/cli/subcommands/check/command.go [hunk 3]
```diff
@@ -200,6 +201,7 @@ func MakeCommand(globalParamsGetter func() GlobalParams) *cobra.Command {
 				fx.Provide(func() inventoryagent.Component { return nil }),
 				fx.Provide(func() inventoryhost.Component { return nil }),
 				fx.Provide(func() packagesigning.Component { return nil }),
+				fx.Provide(func() optional.Option[rcservice.Component] { return optional.NewOption[rcservice.Component](nil) }),
```
Comment (ogaca-dd) [inline pkg/cli/subcommands/check/command.go:204]: You can use `optional.NewNone[rcservice.Component]()`

**Review by ogaca-dd — APPROVED** [review]
LGTM

**Review by ogaca-dd — APPROVED** [review]

**Pull request merged**